kong/vault_service_test.go

```diff
@@ -0,0 +1,206 @@
+package kong
+
+import (
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestVaultsService(t *testing.T) {
```

Comment:
Can we use the cleanup pattern here as well? If any assertion fails (or the test panics between the vault creation and the deletion itself) we'll be left with leftovers in kong's config wouldn't we?